ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/InsertHandler.java

```diff
@@ -81,10 +82,22 @@ public void handle(Context withinContext) throws Exception {
         dataPath = new Path(withinContext.eventRoot, qlPtns.get(0).getName());
       }
 
-      // encoded filename/checksum of files, write into _files
-      try (BufferedWriter fileListWriter = writer(withinContext, dataPath)) {
-        for (String file : files) {
-          fileListWriter.write(file + "\n");
+      int repeat = 0;
+      boolean done = false;
+
+      while (!done) {
+        // encoded filename/checksum of files, write into _files
```

Comment:
Every retry should probably remove the previously created incomplete file.

Reply:
Need to confirm if recreate fails if the file exists and also will it empty the file when we create again.